src/nvim/undo.c

```diff
@@ -1487,7 +1498,83 @@ void u_read_undo(char_u *name, char_u *hash, char_u *orig_name)
   return;
 }
 
+/// Writes a sequence of bytes to the undo file.
+///
+/// @returns false in case of an error.
+static bool undo_write(bufinfo_T *bi, uint8_t *ptr, size_t len)
+{
+  return fwrite(ptr, len, 1, bi->bi_fp) == 1;
+}
+
+/// Writes a number, MSB first, in "len" bytes.
+///
+/// Must match with undo_read_?c() functions.
+///
+/// @returns false in case of an error.
+static bool undo_write_bytes(bufinfo_T *bi, uintmax_t nr, size_t len)
+{
+  assert(len > 0);
+  uint8_t buf[8];
+  for (size_t i = len - 1, bufi = 0; i < len; i--, bufi++) {
+    buf[bufi] = (uint8_t)(nr >> (i * 8));
+  }
+  return undo_write(bi, buf, len);
+}
 
+/*
+ * Write the pointer to an undo header.  Instead of writing the pointer itself
+ * we use the sequence number of the header.  This is converted back to
+ * pointers when reading. */
+static void put_header_ptr(bufinfo_T *bi, u_header_T *uhp)
+{
+  assert(uhp == NULL || uhp->uh_seq >= 0);
+  undo_write_bytes(bi, (uint64_t)(uhp != NULL ? uhp->uh_seq : 0), 4);
+}
+
+static int undo_read_4c(bufinfo_T *bi)
+{
+  return get4c(bi->bi_fp);
+}
+
+static int undo_read_2c(bufinfo_T *bi)
+{
+  return get2c(bi->bi_fp);
+}
+
+static int undo_read_byte(bufinfo_T *bi)
+{
+  return getc(bi->bi_fp);
+}
+
+static time_t undo_read_time(bufinfo_T *bi)
+{
+  return get8ctime(bi->bi_fp);
+}
+
+/// Reads "buffer[size]" from the undo file.
+/// @returns false in case of an error.
+static bool undo_read(bufinfo_T *bi, uint8_t *buffer, size_t size)
+{
+  assert(bi != NULL);
+  return fread(buffer, size, 1, bi->bi_fp) == 1;
+}
+
+/// Reads a string of length "len" from "bi->bi_fd".
+///
+/// Appends a zero to the read string.
+///
+/// @param len can be zero to allocate an empty line.
```

Comment:
What's wrong with that? IMO it's better to be consistent here, because there's big holes in seemingly arbitrary places throughout the doxygen comments. Well maybe not seemingly arbitrary, but it's inconsistent.

On April 6, 2015 11:44:08 PM EDT, Felipe Oliveira Carvalho <notifications@github.com> wrote:
>> +  return get8ctime(bi->bi_fp);
>> +}
>> +
>> +/// Reads "buffer[size]" from the undo file.
>> +/// @returns false in case of an error.
>> +static bool undo_read(bufinfo_T *bi, uint8_t *buffer, size_t size)
>> +{
>> +  assert(bi != NULL);
>> +  return fread(buffer, size, 1, bi->bi_fp) == 1;
>> +}
>> +
>> +/// Reads a string of length "len" from "bi->bi_fd".
>> +///
>> +/// Appends a zero to the read string.
>> +///
>> +/// @param len can be zero to allocate an empty line.
>
>> If one param is going to be documented, then the other should be as
>well.
>
>I don't agree with that because every time I try to follow this
>convention I end up writing really obvious comments about parameters.
>It's better to minimize comments.
>
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/neovim/neovim/pull/1805/files#r27849832